pkg/volume/util/storageclass_test.go

```diff
@@ -0,0 +1,131 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"reflect"
+	"testing"
+	"time"
+
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/kubernetes/pkg/controller"
+)
+
+func TestGetDefaultClass(t *testing.T) {
+
+	var (
+		sc1 *storagev1.StorageClass = &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "my-storage-class1",
+				Annotations: map[string]string{
+					"a": "b",
+				},
+			},
+		}
+		sc2 *storagev1.StorageClass = &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "my-storage-class2",
+				Annotations: map[string]string{
+					"a": "b",
+				},
+			},
+		}
+
+		sc3 *storagev1.StorageClass = &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "my-storage-class3",
+				Annotations: map[string]string{
+					IsDefaultStorageClassAnnotation: "true",
+				},
+				CreationTimestamp: metav1.Time{Time: time.Now().Add(-1 * time.Hour)},
+			},
+		}
+
+		sc4 *storagev1.StorageClass = &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "my-storage-class4",
+				Annotations: map[string]string{
+					IsDefaultStorageClassAnnotation: "true",
+				},
+				CreationTimestamp: metav1.Time{Time: time.Now().Truncate(time.Hour)},
+			},
+		}
+
+		sc5 *storagev1.StorageClass = &storagev1.StorageClass{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "my-storage-class5",
+				Annotations: map[string]string{
+					IsDefaultStorageClassAnnotation: "true",
+				},
+				CreationTimestamp: metav1.Time{Time: time.Now().Truncate(time.Hour)},
+			},
+		}
+	)
+
+	testCases := []struct {
+		name    string
+		classes []*storagev1.StorageClass
+		expect  *storagev1.StorageClass
+	}{
+
+		{
+			name: "no storage class",
+		},
+
+		{
+			name:    "no default storage class",
+			classes: []*storagev1.StorageClass{sc1, sc2},
+			expect:  nil,
+		},
+
+		{
+			name:    "one default storage class",
+			classes: []*storagev1.StorageClass{sc1, sc2, sc3},
+			expect:  sc3,
+		},
+
+		{
+			name:    "three default storage class, newest frist,  class name second",
```

Comment:
split this to two. one for different creationTimestamp. another for different name with same creationTimestamp